students/toddschultz/session04/kata_14/trigrams.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+import random
+
+
+def all_words():
+    all_words = []
+    text = 'great_expectations.txt'
+
+    with open(text, 'r') as f:
+        for item in f:
+            words = item.split()
+            for word in words:
+                word = word.lower()
+                word = word.replace(',', '')
+                word = word.replace('.', '')
+                word = word.replace('?', '')
+                word = word.replace('!', '')
+                word = word.replace('-', '')
+                word = word.replace(' i ', ' qwerty ')
+                if word == 'i':
+                    word = 'I'
+                all_words.append(word)
+        return all_words
+
+def make_trigram():
+    words = all_words()
+    d = dict()
+    for i in range(len(words) - 2):
+        key = tuple(words[i:i+2])
+        value = words[i+2]
+        d.setdefault(key, []).append(value)
+    return d
+
+def make_a_sentence():
+    foo = make_trigram()
+    story = []
+    sentence = (random.choice(list(foo.keys())))
```

Comment:
can be simplified a little: `sentence = random.choice(list(foo))`

also, no foos please 🙂 - remember your code is more often read than written so good variable names are important.